interpreter/src/main/java/org/qbicc/interpreter/impl/VmClassImpl.java

```diff
@@ -121,6 +131,49 @@
         staticLayoutInfo = Layout.getForInterpreter(ctxt).getInterpreterStaticLayoutInfo(typeDefinition);
         staticMemory = staticLayoutInfo == null ? vm.allocate(0) : vm.allocate((int) staticLayoutInfo.getCompoundType().getSize());
         superClass = new VmClassImpl(vm, (VmClassClassImpl) this, classContext.findDefinedType("java/lang/Object").load(), null);
+        initializeConstantStaticFields();
+    }
+
+    void initializeConstantStaticFields() {
+        int cnt = typeDefinition.getFieldCount();
+        for (int i = 0; i < cnt; i ++) {
+            FieldElement field = typeDefinition.getField(i);
+            if (field.isStatic()) {
+                Literal initValue = field.getInitialValue();
+                if (initValue instanceof ZeroInitializerLiteral) {
+                    // Nothing to do;  memory starts zeroed.
+                    continue;
+                }
+                CompoundType.Member member = staticLayoutInfo.getMember(field);
+                if (initValue instanceof IntegerLiteral) {
+                    IntegerLiteral val = (IntegerLiteral) initValue;
+                    if (val.getType().getSize() == 1) {
+                        staticMemory.store8(member.getOffset(), val.byteValue(), MemoryAtomicityMode.UNORDERED);
+                    } else if (val.getType().getSize() == 2) {
+                        staticMemory.store16(member.getOffset(), val.shortValue(), MemoryAtomicityMode.UNORDERED);
+                    } else if (val.getType().getSize() == 4) {
+                        staticMemory.store32(member.getOffset(), val.intValue(), MemoryAtomicityMode.UNORDERED);
+                    } else {
+                        staticMemory.store64(member.getOffset(), val.longValue(), MemoryAtomicityMode.UNORDERED);
+                    }
+                } else if (initValue instanceof FloatLiteral) {
+                    FloatLiteral val = (FloatLiteral) initValue;
+                    if (val.getType().getSize() == 4) {
+                        staticMemory.store32(member.getOffset(), val.floatValue(), MemoryAtomicityMode.UNORDERED);
+                    } else {
+                        staticMemory.store64(member.getOffset(), val.doubleValue(), MemoryAtomicityMode.UNORDERED);
+                    }
+                } else if (initValue instanceof StringLiteral) {
+                    if (vm.bootstrapComplete) {
+                        VmString sv = vm.intern(((StringLiteral) initValue).getValue());
+                        staticMemory.storeRef(member.getOffset(), sv, MemoryAtomicityMode.UNORDERED);
+                    }
+                } else {
+                    // CONSTANT_Class, CONSTANT_MethodHandle, CONSTANT_MethodType
+                    vm.getCompilationContext().warning("Did not properly initialize interpreter memory for constant static field "+field);
```

Comment:
This generates a _ton_ of warnings right now.  Many of them associated with Enums.

Reply:
Fixed; forgot that a null return from getInitialValue means that the field is should just be zero initialized.